File: UI/src/types/expense.ts
/**
 * Expense model matching API structure
 */
export interface Expense {
  id: string; // GUID
  budgetCategoryID: string; // GUID
  amount: number;
  expenseDate: string; // ISO date string
  description: string;
  vendor?: string;
  createdAt?: string;
  updatedAt?: string;
}

/**
 * Request model for creating a new expense
 */
export interface CreateExpenseRequest {
  budgetCategoryID: string;
  amount: number;
  expenseDate: string;
  description: string;
  vendor?: string;
}

/**
 * Expense summary by category
 */
export interface ExpenseSummary {
  categoryID: string;
  categoryName: string;
  totalExpenses: number;
  allocatedAmount: number;
  remainingAmount: number;
  percentageUsed: number;
}

File: API/Models/Expense.cs
namespace BudgetManagement.Api.Models;

public class Expense
{
    public Guid ID { get; set; }
    public Guid BudgetCategoryID { get; set; }
    public decimal Amount { get; set; }
    public DateTime ExpenseDate { get; set; }
    public string? Description { get; set; }
    public string? Vendor { get; set; }
    public string? ReceiptNumber { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public string CreatedBy { get; set; } = string.Empty;
    public string? UpdatedBy { get; set; }
    
    // Navigation properties
    public BudgetCategory? BudgetCategory { get; set; }
}

